code/modules/arcade/mob_hunt/mob_avatar.dm

```diff
@@ -12,17 +11,24 @@
 	var/list/clients_encountered = list()		//tracks who has already interacted with us, so they can't attempt a second capture
 	var/image/avatar
 
-/obj/effect/nanomob/New(loc, datum/mob_hunt/new_info)
-	..()
+/obj/effect/nanomob/Initialize(mapload, datum/mob_hunt/new_info)
+	. = ..()
 	if(!new_info)
-		qdel(src)
-		return
+		return INITIALIZE_HINT_QDEL
 	mob_info = new_info
 	update_self()
 	forceMove(mob_info.spawn_point)
 	if(!mob_info.is_trap)
 		addtimer(CALLBACK(src, .proc/despawn), mob_info.lifetime)
 
+/obj/effect/nanomob/Destroy()
+	SSmob_hunt.trap_spawns -= src
+	SSmob_hunt.normal_spawns -= src
+	clients_encountered.Cut()
+	QDEL_NULL(mob_info)
```

Comment:
```suggestion
	mob_info = null
```
I believe that `mob_info` is given to players who caught him to save it in their `captured` list. Multiple players can catch the same mob as well so `mob_info` is passed around a lot. Deleting it here will cause a null pointer in their reference